lib/matplotlib/axes/_base.py

```diff
@@ -985,14 +1040,30 @@ def clear(self):
         """clear the axes"""
         self.cla()
 
+    def set_prop_cycle(self, prop_cycle):
+        """
+        Set the prop cycle for any future plot commands on this Axes.
+
+        *prop_cycle* is a :class:Cycler object.
+        Can also be `None` to reset to the cycle defined by the
+        current style.
+        """
+        self._get_lines.set_prop_cycle(prop_cycle)
+        self._get_patches_for_fill.set_prop_cycle(prop_cycle)
+
     def set_color_cycle(self, clist):
         """
         Set the color cycle for any future plot commands on this Axes.
 
         *clist* is a list of mpl color specifiers.
+
+        .. deprecated:: 1.5
```

Comment:
So, I want to clear up exactly what we want to deprecate and what we want to keep around. There is the `axes.color_cycle` rcparam and then there is `Axes.set_color_cycle()`. I definitely want to get rid of the rcparam. If both `axes.color_cycle` and `axes.prop_cycle` are in the same rc file, then it is impossible to fully control which one takes priority, so it could be that sometimes the color-cycle wins, and sometimes the property cycle wins (I think the contents of the file gets turned into a dictionary at some point in the process).

I can see the point of keeping around `set_color_cycle()` for a while because the user controls exactly when it calls this method. I should make it clear that it blows away any existing property cyclers.